applications/asset_tracker_v2/src/modules/sensor_module.c

```diff
@@ -263,8 +264,16 @@ static void battery_data_get(void)
 	sensor_module_event->data.bat.timestamp = k_uptime_get();
 	sensor_module_event->data.bat.battery_level = percentage;
 	sensor_module_event->type = SENSOR_EVT_FUEL_GAUGE_READY;
-	APP_EVENT_SUBMIT(sensor_module_event);
+#else
+	sensor_module_event = new_sensor_module_event();
+
+	if (!sensor_module_event) {
+		SEND_ERROR(sensor, SENSOR_EVT_ERROR, -ENOMEM);
```

Comment:
Is there a reason why this is here but not in the `#if defined(CONFIG_ADP536X)` branch?

Reply:
Looking at the `SEND_ERROR()` macro implementation, the first thing it does is to try to allocate a new event by calling `new_sensor_module_event()`. 😃 It does assert that the allocation succeeds, though. So, if asserts are enabled, at least the device would crash with a clearly visible reason.

I'm not sure if the check is needed here, but at least it would be nice to be consistent.

Reply:
> Is there a reason why this is here but not in the `#if defined(CONFIG_ADP536X)` branch?

No, that's probably just an overlook. There should be a check, I'll add it.

> Looking at the `SEND_ERROR()` macro implementation, the first thing it does is to try to allocate a new event by calling `new_sensor_module_event()`. 😃 It does assert that the allocation succeeds, though. So, if asserts are enabled, at least the device would crash with a clearly visible reason.
> 
> I'm not sure if the check is needed here, but at least it would be nice to be consistent.

I did it this way to avoid asserting to also catch it when asserts are available, but need to reconsider that.
I'll just switch back to assert and we'll need to revisit this topic later.